Rubberduck.RibbonDispatcher/AbstractRibbon.cs

```diff
@@ -0,0 +1,67 @@
+    ///    instead of a plain OnAction(,).
+    /// </remarks>
+    [ComVisible(true)][CLSCompliant(true)]
+    public abstract class AbstractRibbon {        
+        public virtual void OnRibbonLoad(IRibbonUI ribbonUI) {
+            RibbonFactory         = new RibbonFactory(ribbonUI);
+        }
+        protected RibbonFactory     RibbonFactory    { get; private set; }    // A property because might be exposed in future
+
+        public IRibbonCommon     Controls(string controlId)              => GetValueOrNull(RibbonFactory.Controls, controlId);
+        public IRibbonButton     Buttons(string controlId)               => GetValueOrNull(RibbonFactory.Buttons,  controlId);
+        public IRibbonToggle     Toggles(string controlId)               => GetValueOrNull(RibbonFactory.Toggles,  controlId);
+        public IRibbonDropDown   DropDowns(string controlId)             => GetValueOrNull(RibbonFactory.DropDowns,controlId);
+
+        // All controls (almost) including Groups
+        public string            GetDescription(IRibbonControl control)  => Controls(control?.Id)?.Description??Unknown(control);
+        public string            GetKeyTip(IRibbonControl control)       => Controls(control?.Id)?.KeyTip     ??"??";
+        public string            GetLabel(IRibbonControl control)        => Controls(control?.Id)?.Label      ??Unknown(control);
+        public string            GetScreenTip(IRibbonControl control)    => Controls(control?.Id)?.ScreenTip  ??Unknown(control);
+        public string            GetSuperTip(IRibbonControl control)     => Controls(control?.Id)?.SuperTip   ??Unknown(control);
+
+        public bool              GetEnabled(IRibbonControl control)      => Controls(control?.Id)?.Enabled    ??false;
+        public IPictureDisp      GetImage(IRibbonControl control)        => Controls(control?.Id)?.Image;
+        public RibbonControlSize GetSize(IRibbonControl control)         => Controls(control?.Id)?.Size       ??RibbonControlSizeLarge;
+        public bool              GetVisible(IRibbonControl control)      => Controls(control?.Id)?.Visible    ??true;
+
+        // Button Controls
+        public bool              GetShowImage(IRibbonControl control)    => Buttons(control?.Id)?.ShowImage   ??false;
+        public bool              GetShowLabel(IRibbonControl control)    => Buttons(control?.Id)?.ShowLabel   ??true;
+        public void              OnAction(IRibbonControl control)        => Buttons(control?.Id)?.OnAction();
+
+        // Toggle Controls: checkBoxes & toggleButtons
+        public bool GetPressed(IRibbonControl control)                   => Toggles(control?.Id)?.IsPressed   ??false;
+        public void OnActionToggle(IRibbonControl control, bool pressed) => Toggles(control?.Id)?.OnAction(pressed);
+
+        private static TValue GetValueOrNull<TValue>(IReadOnlyDictionary<string,TValue> dictionary, string key)
```

Comment:
it's called "GetOrDefault", which is the actual behaviour. You're not enforcing `Null` in any way, which means this is a misleading name for dictionaries that have non-object types as TValue. 

Either enforce `null` explicitly or drop the method.